src/server/server_family.cc

```diff
@@ -328,6 +332,16 @@ ServerFamily::ServerFamily(Service* service) : service_(*service) {
     master_id_ = GetRandomHex(eng, CONFIG_RUN_ID_SIZE);
     DCHECK_EQ(CONFIG_RUN_ID_SIZE, master_id_.size());
   }
+
+  string cluster_mode = GetFlag(FLAGS_cluster_mode);
+  if (cluster_mode.empty()) {
+    is_emulated_cluster_ = false;
+  } else if (cluster_mode == "emulated") {
+    is_emulated_cluster_ = true;
+  } else {
+    LOG(ERROR) << "invalid cluster_mode. Exiting...";
+    throw;
```

Comment:
fyi: we do not use exceptions as a rule. I prefer using either `LOG(FATAL)` or `exit(1)` .